pkg/controllers/core/deployer_controller.go

```diff
@@ -613,8 +636,37 @@ func (r *DeployerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.Service{}).
 		Owns(&networkingv1beta1.Ingress{}).
 		// watch for build mutations to update dependent deployers
+		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, r.enqueueTrackedConfigMapResource()).
 		Watches(&source.Kind{Type: &buildv1alpha1.Application{}}, enqueueTrackedResources(&buildv1alpha1.Application{})).
 		Watches(&source.Kind{Type: &buildv1alpha1.Container{}}, enqueueTrackedResources(&buildv1alpha1.Container{})).
 		Watches(&source.Kind{Type: &buildv1alpha1.Function{}}, enqueueTrackedResources(&buildv1alpha1.Function{})).
 		Complete(r)
 }
+
+func (r *DeployerReconciler) constructHostForDeployer(deployer *corev1alpha1.Deployer, cm corev1.ConfigMap) string {
+	domain := defaultDomain
+	if d, ok := cm.Data[defaultDomainKey]; ok {
+		domain = d
+	}
+
+	return fmt.Sprintf("%s.%s", deployer.Name, domain)
+}
+
+func (r *DeployerReconciler) enqueueTrackedConfigMapResource() handler.EventHandler {
```

Comment:
let's enhance the current  enqueueTrackedResource function to support configmaps rather than redefine it

Reply:
After thinking about this more, the config map resource is different from the other ones. We only want to watch for changes to the config map with a specific name in the system namespace. Whereas the generic `enqueueTrackedResource` tracks all items of the kind passed in.

Reply:
That's the role of the Tracker. The `ToRequests` method will be called for every event for the watched kind. The tracker won't be tracking any resources for most ConfigMaps, so there is nothing to queue. For the target ConfigMap, every deployer will be tracking it, so every deployer will be added to the reconciliation queue.

Reply:
Updated!

Reply:
If a lookup happens for something not being tracked, it will log that the item is not being tracked. The logs get a lot of these messages.